src/hotspot/share/gc/g1/g1CodeRootSet.cpp

```diff
@@ -189,7 +189,8 @@ class G1CodeRootSetHashTable : public CHeapObj<mtGC> {
     // A table with the new size should be at most filled by this factor. Otherwise
     // we would grow again quickly.
     const float WantedLoadFactor = 0.5;
-    size_t min_expected_size = checked_cast<size_t>(ceil(current_size / WantedLoadFactor));
+    assert((current_size / WantedLoadFactor) <= SIZE_MAX, "table overflow");
```

Comment:
Surprisingly, this might not work.  See https://bugs.openjdk.org/browse/JDK-8287052.

Reply:
It looks like for clang we should use -Wimplicit-int-conversion instead of
gcc's "-Wconversion -fno-float-conversion". clang seems to have a much richer
set of warning controls in this area than does gcc.

It looks like for clang we should use -Wimplicit-int-conversion instead of
gcc's "-Wconversion -fno-float-conversion". clang seems to have a much richer
set of warning controls in this area than does gcc.

That way we don't implicitly get -Wimplicit-int-float-conversion, which is
what is triggering the warning mentioned in JDK-8287052.  In this case the
loss of precision leading to that warning does not seem important.

Reply:
In my experiments, values up to 0x4000000000000200 will get pass the assert, but then we should hit a different assert later in round_up_power_of_2().  SIZE_MAX can't be represented exactly as a double, so it gets rounded up, and then that assert doesn't correctly check for overflow.